File: worker/split/mqlsplit.py
#!/usr/bin/python

from sys import *

TAG = '-- '

def write_file(filename, buffer):
    file = open(filename, "w")
    file.write(buffer.strip() + "\n")
    file.close()

def get_filename(line):
    name = line.strip()
    name = name.replace(TAG, "")
    name = name.lower()
    return "mql_" + name + ".sql";

if __name__ == '__main__':

    if len(argv) <= 1:
        print "Error: file needed as argument."
        exit(1);

    # init vars
    buffer   = ''
    filename = ".trash"

    # read entire file
    with open(argv[1]) as file:
        content = file.readlines()

    # process file line by line
    for line in content:
        if line.startswith(TAG):
            # flush buffer into file & restart vars
            write_file(filename, buffer)
            buffer   = ''
            filename = get_filename(line)

        # buffer line
        buffer += line;

    # flush buffer into las file
    write_file(filename, buffer);

File: worker/split/mqlsplit.cpp
#include <vector>
#include <stdio.h>
#include <stdlib.h>
#include <fstream>
#include <iostream>
#include <sstream>
#include <string>
#include <algorithm>


using namespace std;

void writeFile (string filename, string buffer) {
  ofstream file;
  file.open (filename);
  file << buffer << endl;
  file.close();
}

/**
 * Save each code-block in a separated file
 * Block starts with a sql-comment-tag `-- TAG`
 */
int main(int argc, char** argv) {

    if (argc <= 1) {
        printf("Error: file needed as argument.\n");
        exit(EXIT_FAILURE);
    }

    // point to file-name argument
    ifstream infile (argv[1]);

    const string TAG = "-- ";
    string line;
    string current_block;
    vector<string> blocks;

    string buffer;
    string block_name;
    string filename = ".trash";

    while (getline(infile, line)) {

        // is TAG header
        if (line.compare(0, TAG.length(), TAG) == 0) {
            // flush buffer
            writeFile (filename, buffer);

            // get new block name
            block_name = line;
            block_name = line.substr(TAG.length(), line.length());
            transform(block_name.begin(), block_name.end(), block_name.begin(), ::tolower);

            // set new tmp vars
            filename = "mql_" + block_name + ".sql";
            buffer.clear();
        }

        // save line
        buffer += line + "\n";
    }

    // flush last block
    writeFile (filename, buffer);

    return 0;
}
